src/containers/Login/LoginContainer.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+
+import LoginModal from "../../components/login/LoginModal";
+import { userLoginRequest } from "../../actions/LoginAction";
+
+export class LoginContainer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      password: "",
+      isLoading: false,
+      errors: {
+        email: "",
+        password: ""
+      }
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    if (this.state.email.length === 0) {
+      this.setState({
+        errors: {
+          email: "The email is required",
+          password: ""
+        }
+      });
+    }
+    const data = {
+      user: {
+        email: this.state.email,
+        password: this.state.password
+      }
+    };
+    this.props.userLoginRequest(data, this.props);
+  }
+
+  render() {
+    return (
+      <div>
+        <LoginModal
+          handleChange={this.handleChange}
+          email={this.state.email}
+          isProcessing={this.props.isProcessing}
+          password={this.state.password}
+          errors={this.state.errors}
+          handleSubmit={this.handleSubmit}
+        />
+      </div>
+    );
+  }
+}
+
+export const mapStateToProps = state => ({
```

Comment:
Why is `mapStateToProps` being exported?

Reply:
Seen why

Reply:
`mapStateToProps` is being exported so that it can be tested in the `LoginContainer.test.js`